src/BCrypt.Tests/BCrypt.cs

```diff
@@ -123,6 +123,84 @@ public void EncryptDecrypt_NoPadding()
         }
     }
 
+    /// <summary>
+    /// Demonstrates use of an authenticated block chaining mode
+    /// that requires use of several more struct types than
+    /// the default CBC mode for AES.
+    /// </summary>
+    [Fact]
+    public unsafe void EncryptDecrypt_AesCcm()
+    {
+        var random = new Random();
+
+        using (var provider = BCryptOpenAlgorithmProvider(AlgorithmIdentifiers.BCRYPT_AES_ALGORITHM))
+        {
+            BCryptSetProperty(provider, PropertyNames.ChainingMode, ChainingModes.Ccm);
+
+            byte[] plainText;
+            byte[] cipherText;
+
+            var nonceBuffer = new byte[12];
+            random.NextBytes(nonceBuffer);
+
+            var tagLengths = BCryptGetProperty<BCRYPT_AUTH_TAG_LENGTHS_STRUCT>(provider, PropertyNames.AuthTagLength);
+            var tagBuffer = new byte[tagLengths.MaxLength];
+
+            int blockSize = BCryptGetProperty<int>(provider, PropertyNames.BlockLength);
+            plainText = new byte[blockSize];
+            random.NextBytes(plainText);
+
+            byte[] keyMaterial = new byte[blockSize];
+            RandomNumberGenerator.Create().GetBytes(keyMaterial);
+
+            using (var key = BCryptGenerateSymmetricKey(provider, keyMaterial))
+            {
+                var authInfo = BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO.Create();
+                fixed (byte* pTagBuffer = &tagBuffer[0])
+                fixed (byte* pNonce = &nonceBuffer[0])
+                {
+                    authInfo.pbNonce = new IntPtr(pNonce);
```

Comment:
Still feel bad to have `IntPtr` instead of pointers here, i'm still conflicted on the subject. Using pointers would block a few esoteric languages (PowerShell is the one with the biggest user base I can think of) but not using it is forcing all callers to copy paste the same conversion.

On that subject I remarket that the current design for [`System.Span`](https://github.com/dotnet/corefxlab/blob/master/src/System.Slices/src/System/Span.cs) only exposes the pointer (`UnsafePointer`) but not an `IntPtr` version, maybe there's a trend starting (Or it's just because the lib development is still in flux and it will change).